oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/OssLicensesPlugin.groovy

```diff
@@ -31,6 +31,12 @@ class OssLicensesPlugin implements Plugin<Project> {
         getDependencies.outputDir = dependencyOutput
         getDependencies.outputFile = generatedJson
 
+        // Collect all build.gradle files used in this project (and subprojects) to be used
```

Comment:
Dependencies and dependency versions may be defined in `gradle.properties` files or environmental variables so I don't think this covers that. They may also be definied in `buildSrc` or composite builds.

Reply:
Do you think it would make sense to add the `gradle.properties` (+ `settings.gradle`?) and `buildSrc` as `@InputFiles` and env maybe as String to be also used as cache key? Or will there be always another way to configure dependencies, so that they are not covered and this "file based" approach won't work in general?
I also have never used composite builds and no idea how to cover that.

I came to this approach when I profiled a bit and saw that most of the time (everything else is done in milliseconds) is used in `updateDependencyArtifacts()` which walks through all dependencies, so I wanted to avoid that for the cache up-to-date check.

Reply:
I think if we're trying to make the dependencies be the input, then we should use the dependencies as inputs. :) 
Using a proxy such as `build.gradle` and `gradle.properties` files is not the same thing. Even if you add all the gradle.properties files, one could load the dependency versions from an arbitrarily names `properties` file or pass the dependency versions either as gradle properties or environment variables. Dependencies can also be definied in composite builds. Covering all the bases here would simply lead to too many things being declared as inputs and making this task never up to date.

Reply:
> I think if we're trying to make the dependencies be the input, then we should use the dependencies as inputs.

Sure, but the problem is that collecting these takes longer than the task execution itself, which would make the caching superfluous.
But I get your point that using the configuration files as proxy doesn't always work. So I'll close this PR and try to work around by disabling this plugin for development builds.

Reply:
Did you ever come up with the workaround for this @schuenadel? Curious if you did as I wasn't able to come up with something on my own, thanks!

Reply:
Looks like i was just able to do this by disabling the task in my debug builds like so:

```
buildTypes {
  getByName("debug") {
    afterEvaluate {
        tasks.withType<com.google.android.gms.oss.licenses.plugin.DependencyTask> {
          enabled = false
        }
      }
   }
}
```

Reply:
I didn't investigate further, but disabling the task for debug builds should be sufficient to eliminate the additional build time for the most frequent builds. And if a release build takes 1-2s longer... 🤷

Reply:
With our app it seems to take 1-2 minutes longer on average and I've seen as high as 276 seconds to run the the `app:getDependencies` task :( For those interested, this is the updated changes we made to disable the task in debug builds (located within our app build.gradle.kts):
```
afterEvaluate {
  tasks.withType<com.google.android.gms.oss.licenses.plugin.DependencyTask> {
    enabled = gradle.startParameter.taskRequests.toString().contains("Release")
  }
}
```